### PR TITLE
feat: add search provider hooks

### PR DESCRIPTION
**Pull request by capJavert**
### Describe what this PR does
- implement search provider hooks 

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

AS-46 #done

### CODE REVIEW

File: packages/shared/src/hooks/search/useSearchProviderSuggestions.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+export type UseSearchProviderSuggestions = {
+  isLoading: boolean;
+  suggestions: Awaited<ReturnType<UseSearchProvider['getSuggestions']>>;
+};
+
+export const useSearchProviderSuggestions = ({
+  provider,
+  query,
+}: UseSearchProviderSuggestionsProps): UseSearchProviderSuggestions => {
```
[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/src/hooks/search/useSearchProviderSuggestions.ts:21]: Yeah the rename sounds good to me :)

File: packages/shared/src/graphql/search.ts [hunk 2]
```diff
@@ -289,3 +289,13 @@ export const sendSearchQuery = async (
 
   return new EventSource(`${searchQueryUrl}?${params}`);
 };
+
+export type SearchSuggestion = {
```
Comment (rebelchris) [inline packages/shared/src/graphql/search.ts:293]: Just nitpicking why this is under the graphql layer, but probably just what we had already?

**Review by rebelchris — APPROVED** [review]
Nothing blocking from my side :)

File: packages/shared/src/graphql/search.ts (hunk 2 above)
Comment (capJavert) [reply packages/shared/src/graphql/search.ts:293]: Its result from search suggestions query, from what I have seen there was no static type for it, only runtime in different places.

File: packages/shared/src/hooks/search/useSearchProviderSuggestions.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (capJavert) [reply packages/shared/src/hooks/search/useSearchProviderSuggestions.ts:21]: PR https://github.com/dailydotdev/apps/pull/2545

**Pull request merged**